corehq/messaging/smsbackends/trumpia/models.py

```diff
@@ -64,6 +64,9 @@ def handle_response(self, response, msg):
             data = self.get_message_details(data["requestID"])
             if is_success(data):
                 return  # success
+            if data.get("message") == "In progress":
+                msg.set_status_pending()
```

Comment:
This will be called once per message (assuming each message is "In progress").

Reply:
can you share when this is done?
Reading from the code it appears to be happening when the SMS is sent so this would happen always?
and then we leave this as status pending always?

Reply:
Yes, this gets set when the message is sent, and it will not updated after that, so the message will remain in this pending state indefinitely. This is because HQ has no mechanism of retrieving the pending status from Trumpia whenever it eventually becomes available. Implementing that would be a bigger lift. As documented in the `get_message_details` docstring, Trumpia can be manually queried for the status of the message on an as-needed basis to troubleshoot issues if it is believed the messages are not being received.

Most gateways immediately respond with a more helpful status that indicates various error conditions, but it appears Trumpia does some sort of async/queue thing where the status is not available for some time after we submit it to their API. From our perspective the message was sent. If it turns out that there are not enough funds in the account or some other issue that prevents Trumpia from sending the message then the account holder will need to monitor for those conditions on the Trumpia side. I would expect Trumpia has error reports that can be used for that purpose.